src/runtime/job.cpp

```diff
@@ -1286,18 +1316,35 @@ static PRIMFN(prim_job_virtual) {
 }
 
 static PRIMTYPE(type_job_create) {
-  return args.size() == 12 && args[0]->unify(Data::typeString) &&
+  return args.size() == 14 && args[0]->unify(Data::typeString) &&
          args[1]->unify(Data::typeString) && args[2]->unify(Data::typeString) &&
          args[3]->unify(Data::typeString) && args[4]->unify(Data::typeString) &&
          args[5]->unify(Data::typeInteger) && args[6]->unify(Data::typeString) &&
          args[7]->unify(Data::typeInteger) && args[8]->unify(Data::typeString) &&
          args[9]->unify(Data::typeString) && args[10]->unify(Data::typeString) &&
-         args[11]->unify(Data::typeInteger) && out->unify(Data::typeJob);
+         args[11]->unify(Data::typeString) && args[12]->unify(Data::typeString) &&
+         args[13]->unify(Data::typeInteger) && out->unify(Data::typeJob);
+}
+
+std::vector<std::string> null_sep_list(const std::string &str) {
```

Comment:
A function like this already exists further into the job.cpp code that is used on cmd and env that you can use instead.

Reply:
Ooh, there is?  Great, thanks!  (I thought I had searched for it, but apparently I did so particularly poorly.)